src/parser.jl

```diff
@@ -1282,9 +1290,9 @@ function parse_unary(ps::ParseState)
                             (initial_semi && num_subexprs > 0)    ||
                             (initial_semi && num_semis == 1)      ||
                             (num_semis == 0 && num_subexprs == 0)
-            return (needs_parameters=is_paren_call,
-                    is_paren_call=is_paren_call,
-                    is_block=!is_paren_call && num_semis > 0)
+            return ParseUnaryBracketOpts(; needs_parameters=is_paren_call,
+                                         is_paren_call=is_paren_call,
+                                         is_block=!is_paren_call && num_semis > 0)
         end
```

Comment:
Does this do the trick?

```suggestion
            return (needs_parameters=is_paren_call,
                    is_paren_call=is_paren_call,
                    is_block=!is_paren_call && num_semis > 0)
        end
        opts::NamedTuple{(:needs_parameters, :is_paren_call, :is_block), Tuple{Bool, Bool, Bool}}
```